packages/runtime-core/src/apiWatch.ts

```diff
@@ -234,17 +234,26 @@ function traverse(value: unknown, seen: Set<unknown> = new Set()) {
       traverse(value[i], seen)
     }
   } else if (value instanceof Map) {
-    value.forEach((v, key) => {
-      // to register mutation dep for existing keys
-      traverse(value.get(key), seen)
-    })
+    const keys = [...value.keys()]
```

Comment:
The cost here is almost the same because it involves spreading the iterator into a plain array. The array spread can often be deceivingly expensive. Not really worth it IMO.

Reply:
On my side I see no significant difference even with a million items in the set. I've looked at both timing and memory performance indicators, refreshing the page for each particular method (so run test with spread, change code to for..of and refresh, a few times over).

Chrome Version 79.0.3945.88 (Official Build) (64-bit) on MacOS Catalina (10.14.6) (spec 2.4GHz i5, 16GB 2133MHz LPDDR3) by launching chrome with:

`/Applications/Google\ Chrome.app/Contents/MacOS/Google\ Chrome --enable-precise-memory-info --js-flags="--trace-opt --trace-deopt"`

```
function forOfSet(val, index) {
  console.time('for set' + index)
  const acc = []
  // const arr = [...val]
  for (const x of val) acc.push(x)
  // for (let i=0; i<arr.length; i++) acc.push(arr[i])
  console.timeEnd('for set' + index)
  console.log('memory: total, used, sizeLimit:', window.performance.memory.totalJSHeapSize, window.performance.memory.usedJSHeapSize, window.performance.memory.jsHeapSizeLimit)
  console.log(acc)
}

const set1 = new Set([1,2,3])
const set2 = new Set(Array(100).fill(null).map((_,i) => i))
const set3 = new Set(Array(1000).fill(null).map((_,i) => i))
const set4 = new Set(Array(1000000).fill(null).map((_,i) => i))

// forOfSet(set1, 1)
// forOfSet(set2, 2)
// forOfSet(set3, 3)
forOfSet(set4, 4)
```

The timing varies with each refresh in a range of 30 to 45 ms, regardless of the loop method being used.

![Screen Shot 2019-12-27 at 5 14 45 PM](https://user-images.githubusercontent.com/264986/71508813-6881d500-28cc-11ea-900f-b1e626b16b8a.png)